storage/netstore.go

```diff
@@ -201,10 +205,10 @@ func (n *NetStore) Get(ctx context.Context, mode chunk.ModeGet, req *Request) (c
 			if ok {
 				ch, err = n.RemoteFetch(ctx, req, fi)
 				if err != nil {
-					if n.recoveryCallback != nil {
-						n.recoveryCallback(ctx, ref)
-						time.Sleep(500 * time.Millisecond) // TODO: view what the ideal timeout is
-						return n.RemoteFetch(ctx, req, fi)
+					if n.recoveryCallback != nil && publisher != "" {
+						log.Debug("content recovery callback triggered", "ref", ref.String())
+						go n.recoveryCallback(ctx, ref)
```

Comment:
based on some tests, we think it might be necessary to call the recovery callback with `go` here.

do you see any issues with this @zelig?